### PR TITLE
Docs: split up reviewing into new file sections

### PR DESCRIPTION
**Pull request by prince-chrismc**
Docs!

- moved attributes + license  stuff to `conanfile_attributes.md`
   - removed duplicated docs from hooks and FAQs  
- fixed spelling in FAQs
- Removed stale documentation around `conans.tools.CMake` (this will come in a future PR under #13673)
- Revamped the `test_packages` section with it's own file

## This one you can give all the feedback in the world

This does not have any "policy" stuff so I am willing to make changes 😸

### CODE REVIEW

File: docs/adding_packages/test_packages.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+|               +-- ...
+|           +-- test_cmakedeps/
+|               +-- conanfile.py
+|               +-- CMakeLists.txt
+|               +-- test_package.cpp
+```
+
+### Testing CMake variables from FindModules
+
+@prince-chrism write docs here.
```
Comment (SSE4) [inline docs/adding_packages/test_packages.md:73]: leftover?

Comment (prince-chrismc) [reply docs/adding_packages/test_packages.md:73]: 🙈 Yes

**Comment by prince-chrismc** [comment]
https://github.com/conan-io/conan-center-index/actions/runs/3595997793/jobs/6056156046#step:4:92

are fixed with #14523

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**